play24/src/main/scala/com/gu/cm/ContextConfigurationLoader.scala

```diff
@@ -0,0 +1,17 @@
+package com.gu.cm
+
+import play.api.{Configuration => PlayConfiguration}
+import play.api.ApplicationLoader.Context
+import PlayImplicits._
+
+object ContextConfigurationLoader {
+  def apply(defaultAppName: String, context: Context): Context = {
+    val config = Configuration(
+      defaultAppName = defaultAppName,
+      mode = context.environment.mode,
+      logger = PlayDefaultLogger
+    ).load
```

Comment:
It's best to call `resolve` after loading. I had a problem because my config file included a placeholder:

```
play.crypto.secret = "changeme"
play.crypto.secret = ${?PLAY_CRYPTO_SECRET}
```

This caused an exception to be thrown at startup (either when wrapping it in `PlayConfiguration` or when calling `++`, I can't remember which).